File: 0443-string-compression/0443-string-compression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        int n = chars.size();
        int read =0;
        int write =0;
        while(read < n){
            char curr = chars[read];
            int count =0;

            while(read < n && chars[read] == curr){
                read++;
                count++;
            }

            chars[write++] = curr;

            if(count > 1){
                string temp = to_string(count);
                for(char c: temp) chars[write++] = c;
            }
        }
        return write;
    }
};

File: 0443-string-compression/0443-string-compression.java
class Solution {
    public int compress(char[] chars) {
        int n = chars.length;
        int start =0,move =0;
        while(move < n){
            char curr = chars[move];
            int count = 0;
            
            while(move < n && chars[move] == curr){
                move++;
                count++;
            }

            chars[start++] = curr;

            if(count > 1){
                String temp = Integer.toString(count);
                for(char c: temp.toCharArray()){
                    chars[start++] = c;
                }
            }
        }
        return start;
    }
}